packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts

```diff
@@ -0,0 +1,127 @@
+
+export const getStyles = memoizeFunction((
+  theme: ITheme,
+  customStyles?: ICheckboxStyles
+): ICheckboxStyles => {
+  const { semanticColors, palette } = theme;
+  const checkmarkFontColor = semanticColors.inputForegroundChecked;
+  const checkboxBorderColor = semanticColors.inputBorder;
+  const checkboxBorderHoveredColor = semanticColors.inputBorderHovered;
+  const checkboxBackgroundChecked = semanticColors.inputBackgroundChecked;
+  const checkboxBackgroundCheckedHovered = semanticColors.inputBackgroundCheckedHovered;
+  const checkboxBackgroundDisabled = semanticColors.disabledText;
+  const checkboxTextColorDisabled = semanticColors.disabledText;
+
+  const styles: ICheckboxStyles = {
+    root: [
+      getFocusStyle(theme),
+      {
+        padding: '0',
+        border: 'none',
+        background: 'none',
+        margin: '0',
+        outline: 'none',
+        display: 'block',
+        cursor: 'pointer',
+
+        // This is to hide the label's overflow because of the negative margin.
+        overflow: 'hidden'
+      }
+    ],
+    label: {
+      display: 'inline-flex',
+      alignItems: 'center',
+      cursor: 'pointer',
+
+      // The negative margin is in place to correct for the margin of the box
+      margin: '0 -4px',
+      position: 'relative',
+      userSelect: 'none'
+    },
+    labelReversed: {
+      flexDirection: 'row-reverse',
+      justifyContent: 'flex-end'
+    },
+    labelDisabled: {
+      cursor: 'default'
+    },
+    input: {
+      position: 'absolute',
+      opacity: '0'
+    },
+    box: {
+      display: 'flex',
+      margin: '0 4px',
+      alignItems: 'center',
+      justifyContent: 'center',
+      flex: '0 0 ' + MS_CHECKBOX_LABEL_SIZE,
+      height: MS_CHECKBOX_LABEL_SIZE,
+    },
+    checkbox: {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      height: MS_CHECKBOX_LABEL_SIZE,
+      width: MS_CHECKBOX_LABEL_SIZE,
+      borderWidth: '1px',
+      borderStyle: 'solid',
+      borderColor: checkboxBorderColor,
+      boxSizing: 'border-box',
+      transitionProperty: 'background, border, border-color',
+      transitionDuration: MS_CHECKBOX_TRANSITION_DURATION,
+      transitionTimingFunction: MS_CHECKBOX_TRANSITION_TIMING,
+
+      /* incase the icon is bigger than the box */
+      overflow: 'hidden',
+    },
+    checkboxHovered: {
+      borderColor: checkboxBorderHoveredColor,
+    },
+    checkboxChecked: {
+      background: checkboxBackgroundChecked,
+    },
+    checkboxCheckedHovered: {
+      background: checkboxBackgroundCheckedHovered,
+    },
+    checkboxDisabled: {
+      background: checkboxBackgroundDisabled,
+    },
+    checkboxCheckedDisabled: {
+      background: checkboxBackgroundDisabled,
+    },
+    checkmark: {
+      opacity: '0',
+      color: checkmarkFontColor
+    },
+    checkmarkChecked: {
+      opacity: '1'
+    },
+    checkmarkDisabled: {
+    },
+    checkmarkCheckedDisabled: {

```

Comment:
What it was before is incorrect because we want disabled colors to be paired with disable colors, input colors paired with input colors, and not mix between the two.